sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatRequestUserMessage.java

```diff
@@ -42,48 +41,18 @@ public final class ChatRequestUserMessage extends ChatRequestMessage {
      */
     @Generated
     @JsonCreator
-    public ChatRequestUserMessage(@JsonProperty(value = "content") BinaryData content) {
+    public ChatRequestUserMessage(@JsonProperty(value = "content") List<ChatMessageContentItem> content) {
         this.content = content;
     }
```

Comment:
Recommend not touch anything about `@JsonXXX` as they are involved with serialization/de-serialization

have another ctor to handle the conversion, e.g.

```java
    private ChatRequestUserMessage(@JsonProperty(value = "content") BinaryData content) {
        this.content = content;
    }

    public ChatRequestUserMessage(List<ChatMessageContentItem> content) {
        this(BinaryData.fromObject(content));
    }
```

Reply:
OK, this does not work in runtime. The serialized data misses the `type` discriminator.